pyvortex/src/array.rs

```diff
@@ -138,6 +138,71 @@ impl PyArray {
         PyDType::wrap(self_.py(), self_.inner.dtype().clone())
     }
 
+    /// Point-wise compare the elements of this array to another array.
+    ///
+    /// Parameters
+    /// ----------
+    /// other : :class:`vortex.encoding.Array`
+    ///     An array with whom to compare elements.
+    ///
+    /// operator : :class:`str`
+    ///     One of `eq`, `ne`, `gt`, `ge`, `lt`, or `le` indicating which binary comparison operator
+    ///     to apply.
+    ///
+    /// Returns
+    /// -------
+    /// :class:`vortex.encoding.Array`
+    ///
+    /// Examples
+    /// --------
+    ///
+    /// Compare an array of strings to itself:
+    ///
+    /// >>> a = vortex.encoding.array(['a', 'b', 'c', 'd'])
+    /// >>> a.compare(a, "eq").to_arrow_array()
+    /// <pyarrow.lib.BooleanArray object at ...>
+    /// [
+    ///   true,
+    ///   true,
+    ///   true,
+    ///   true
+    /// ]
+    ///
+    /// Compare two arrays containing nulls:
+    ///
+    /// >>> a = vortex.encoding.array(['dog', None, 'cat', 'mouse', 'fish'])
+    /// >>> b = vortex.encoding.array(['doug', 'jennifer', 'casper', 'mouse', 'faust'])
+    /// >>> a.compare(b, 'lt').to_arrow_array()
+    /// <pyarrow.lib.BooleanArray object at ...>
+    /// [
+    ///   true,
+    ///   null,
+    ///   false,
+    ///   false,
+    ///   false
+    /// ]
+    fn compare(&self, other: &Bound<PyArray>, operator: &Bound<PyString>) -> PyResult<PyArray> {
```

Comment:
I think this is a very weird python api, would this be better as a individual funtions instead of string switch?

Reply:
I agree it is weird. PyArrow doesn't implement __lt__ and friends but, AFAICT, nobody has complained on their issues about it. They do implement `__eq__`. In the compute functions API, they do provide [`greater`](https://arrow.apache.org/docs/python/generated/pyarrow.compute.greater.html) and friends.

My best guess is that those operations are allocating so they make them available through an API that exposes the memory pool parameter. It seems like that's [a way to use](https://arrow.apache.org/docs/cpp/memory.html#default-memory-pool) jemalloc or mimalloc for Arrow arrays even if the rest of the application uses some other allocator. It seems like that could be used to create an Arena for arrow arrays and do O(1) free after, say, processing a chunk of data.

Anyway, I think you're right that we should implement the operators. Even if we later provide a way to specify an allocator, the operators can continue to use the default allocator.

Reply:
I was mostly even thinking that even if it's not the `__eq__` or `__rich_compare__` I think it's better to get rid of stringly typed operator

Reply:
I switched to operators and moved the docs into the examples for technical reasons (see comment).